### PR TITLE
Updated docs

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John <john.mccrae@progress.com>

## Description

Substantially updating the Chef Client documents to reflect Chef-18 changes, fixes, et al

## Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussion that's relevant]

## Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

### CODE REVIEW

File: data/infra/resources/batch.yaml [hunk 1]
```diff
@@ -143,7 +209,7 @@ properties_list:
   ruby_type: String
   required: false
   description_list:
-  - markdown: '*Windows only*: The password of the user specified by the <span
+  - markdown: 'On Windows only, if this property is specified, the
```
Comment (IanMadd) [inline data/infra/resources/batch.yaml]: I'm not sure what the text should be here.

**Comment by IanMadd** [comment]
Does this handle the issue in chef/chef#13166? It looks like, for example, the snap_package actions should only be :install, :upgrade, :remove, :purge and exclude lock, reconfig, and unlock.

**Comment by johnmccrae** [comment]
Ian, didn't even catch that. Docs now reflect that PR.

**Comment by johnmccrae** [comment]
@IanMadd Docs-tag! You're it!

**Comment by Stromweld** [comment]
ooh I've been waiting to see these resource docs updated.

**Comment by johnmccrae** [comment]
@Stromweld I almost borked it and didn't add your changes. Ian caught me. Lemme know if I missed anything

**Comment by Stromweld** [comment]
The few that I was looking for are in there. Thanks.

**Comment by IanMadd** [comment]
Is the bash resource correct? It's got text referencing the execute resource. [data/infra/resources/bash.yaml](https://github.com/chef/chef-web-docs/pull/4004/files#diff-3dff777e7577c3c73489a5123bd3f88171948d32d0ba6ac0b1e9cc9fa70eabdc)

> Use the **execute** resource to run a single command. Use multiple
        **execute** resource blocks to run multiple commands.

File: data/infra/resources/batch.yaml [hunk 2]
```diff
@@ -91,20 +97,13 @@ properties_list:
...
+  - markdown: 'When this property is set to :batch, the 64-bit version of the cmd.exe shell will be used to evaluate strings values for the not_if and only_if properties.
+      Set this value to :default to use the 32-bit version of the cmd.exe shell.'
 - property: interpreter
   ruby_type: String
   required: false
   description_list:
-  - markdown: 'The script interpreter to use during code execution. Changing the
-
-      default value of this property is not supported.'
+  - markdown:
```
Comment (IanMadd) [inline data/infra/resources/batch.yaml]: Empty string here

File: data/infra/resources/batch.yaml [hunk 3]
```diff
@@ -119,71 +118,29 @@ properties_list:
   description_list:
   - markdown: The amount of time (in seconds) a command is to wait before timing out.
 - property: user
-  ruby_type: String
+  ruby_type: String, Integer
```
Comment (IanMadd) [inline data/infra/resources/batch.yaml]: Is this right?

File: data/infra/resources/bash.yaml [hunk 4]
```diff
@@ -99,74 +169,126 @@ properties_list:
...
 
-      command.'
-- property: umask
-  ruby_type: String, Integer
-  required: false
-  description_list:
-  - markdown: The file mode creation mask, or umask.
-resources_common_properties: true
-resources_common_notification: true
-resources_common_guards: true
```
Comment (IanMadd) [inline data/infra/resources/bash.yaml:112]: These were all undone on all the resources that were updated. Is that correct?

File: data/infra/resources/cookbook_file.yaml [hunk 5]
```diff
@@ -83,13 +71,31 @@ properties_list:
...
   required: false
   default_value: The current cookbook name
   description_list:
   - markdown: The cookbook in which a file is located (if it is not located in the
       current cookbook).
+- property: deny_rights
+  ruby_type: ''
+  required: false
+  description_list:
+  - markdown:
```
Comment (IanMadd) [inline data/infra/resources/cookbook_file.yaml]: Are these properties correct?

File: data/infra/resources/csh.yaml [hunk 6]
```diff
@@ -10,39 +10,58 @@ resource_description_list:
...
+- property: command
+  ruby_type: String, Array
+  required: false
+  default_value: The resource block's name.
+  description_list:
+  - markdown: An optional property to set the command to be executed if it differs
+      from the resource block's name.
+  - note:
+      markdown: Use the **execute** resource to run a single command. Use multiple
+        **execute** resource blocks to run multiple commands.
```
Comment (IanMadd) [inline data/infra/resources/csh.yaml]: Is this correct?

File: data/infra/resources/csh.yaml [hunk 7]
```diff
@@ -53,6 +72,36 @@ properties_list:
...
+- property: elevated
+  ruby_type: true, false
+  required: false
+  default_value: 'false'
+  new_in: '13.3'
+  description_list:
+  - markdown: |-
+      Determines whether the script will run with elevated permissions to circumvent User Access Control (UAC) from interactively blocking the process.
+      This will cause the process to be run under a batch login instead of an interactive login. The user running chef-client needs the 'Replace a process level token' and 'Adjust Memory Quotas for a process' permissions. The user that is running the command needs the 'Log on as a batch job' permission.
+      Because this requires a login, the user and password properties are required.
```
Comment (IanMadd) [inline data/infra/resources/csh.yaml]: Is this correct?

File: data/infra/resources/csh.yaml [hunk 8]
```diff
@@ -71,6 +120,41 @@ properties_list:
...
+      execution environment.
+- property: password
+  ruby_type: String
+  required: false
+  new_in: '12.21'
+  description_list:
+  - markdown: 'Windows only: The password of the user specified by the user property.
+      This property is mandatory if user is specified on Windows and may only be specified
+      if user is specified. The sensitive property for this resource will automatically
+      be set to true if password is specified.'
```
Comment (IanMadd) [inline data/infra/resources/csh.yaml]: Also this?

File: data/infra/resources/directory.yaml [hunk 9]
```diff
@@ -156,79 +162,7 @@ properties_list:
...
-
-      Microsoft Windows environment. For example:
-
-      `rights <permissions>, <principal>, <options>` where `<permissions>`
-
-      specifies the rights granted to the principal, `<principal>` is the
-
-      group or user name, and `<options>` is a Hash with one (or more)
-
-      advanced rights options.'
```
Comment (IanMadd) [inline data/infra/resources/directory.yaml]: Should this be deleted?

File: data/infra/resources/dmg_package.yaml [hunk 10]
```diff
@@ -39,8 +39,6 @@ syntax_full_properties_list:
   `dmg_passphrase`, `file`, `headers`, `owner`, `package_id`, `source`, `type`, and
   `volumes_dir` are the properties available to this resource."
 actions_list:
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
   :install:
```
Comment (IanMadd) [inline data/infra/resources/dmg_package.yaml:42]: should this be deleted?

File: data/infra/resources/dnf_package.yaml [hunk 11]
```diff
@@ -50,22 +45,26 @@ syntax_full_properties_list:
...
     markdown: Purge a package. This action typically removes the configuration files
       as well as the package.
+  :reconfig:
+    markdown: Change the installed package
+  :lock:
+    markdown: Locks the DNF package to a specific version.
   :unlock:
     markdown: Unlocks the DNF package so that it can be upgraded to a newer version.
+  :flush_cache:
+    markdown:
```
Comment (IanMadd) [inline data/infra/resources/dnf_package.yaml]: missing text

File: data/infra/resources/dsc_resource.yaml [hunk 12]
```diff
@@ -5,192 +5,98 @@ resources_common_notification: true
 resources_common_properties: true
 resource: dsc_resource
 resource_description_list:
-- shortcode: resources_common_powershell.md
-- shortcode: resources_common_powershell_dsc.md
```
Comment (IanMadd) [inline data/infra/resources/dsc_resource.yaml]: Should these be deleted?

File: data/infra/resources/dsc_resource.yaml [hunk 13]
```diff
@@ -378,9 +284,6 @@ properties_list:
   - markdown: 'The amount of time (in seconds) a command is to wait before timing
 
       out.'
-resources_common_properties: true
-resources_common_notification: true
-resources_common_guards: true
```
Comment (IanMadd) [inline data/infra/resources/dsc_resource.yaml:383]: Should these be deleted?

File: data/infra/resources/directory.yaml (hunk 9 above)
Comment (Stromweld) [reply data/infra/resources/directory.yaml]: Yeah something is weird here. This shouln't be deleted.

Comment (Stromweld) [reply data/infra/resources/directory.yaml]: Currently published docs are also missing deny_rights property but it is talked about in the windows security section. https://docs.chef.io/resources/directory/#:~:text=Use%20the-,deny_rights,-property%20to%20deny

File: data/infra/resources/dsc_script.yaml [hunk 14]
```diff
@@ -6,72 +6,35 @@ resources_common_notification: true
...
-
-    configuration directly within a Chef Infra recipe.'
-- note:
-    markdown: 'Windows PowerShell 4.0 is required for using the **dsc_script**
-
-      resource with Chef Infra.'
-- note:
-    markdown: 'The WinRM service must be enabled. (Use `winrm quickconfig` to enable
-
-      the service.)'
```
Comment (IanMadd) [inline data/infra/resources/dsc_script.yaml]: Should these be deleted?

File: data/infra/resources/file.yaml [hunk 15]
```diff
@@ -124,126 +117,25 @@ properties_list:
...
-
-      and content of symlink''s target file.
-
-
-      The default behavior is `true` but emits a warning that the default
-
-      value will be changed to `false` in a future version; setting this
-
-      explicitly to `true` or `false` suppresses this warning.'
-- property: mode
```
Comment (IanMadd) [inline data/infra/resources/file.yaml:150]: It looks like the mode and owner properties are still valid.

File: data/infra/resources/file.yaml [hunk 16]
```diff
@@ -88,6 +76,11 @@ properties_list:
   - markdown: A string that is written to the file. The contents of this property
       replace any previous content when this property has something other than the
       default value. The default behavior will not modify content.
+- property: deny_rights
+  ruby_type: ''
+  required: false
+  description_list:
+  - markdown:
```
Comment (IanMadd) [inline data/infra/resources/file.yaml]: Missing text and a ruby type.

File: data/infra/resources/habitat_package.yaml [hunk 17]
```diff
@@ -34,24 +34,18 @@ syntax_full_properties_list:
   `options`, `package_name`, `source`, `timeout`, and `version` are the properties
   available to this resource."
 actions_list:
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
```
Comment (IanMadd) [inline data/infra/resources/habitat_package.yaml:38]: Should this be deleted?

File: data/infra/resources/homebrew_package.yaml [hunk 18]
```diff
@@ -32,18 +32,16 @@ syntax_full_properties_list:
 - "`homebrew_user`, `options`, `package_name`, `source`, `timeout`, and `version`
   are the properties available to this resource."
 actions_list:
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
   :install:
```
Comment (IanMadd) [inline data/infra/resources/homebrew_package.yaml:35]: Should this be deleted?

File: data/infra/resources/launchd.yaml [hunk 19]
```diff
@@ -82,24 +82,29 @@ syntax_full_properties_list:
...
+  :nothing:
+    shortcode: resources_common_actions_nothing.md
   :restart:
     markdown: Restart a launchd managed daemon or agent.
 properties_list:
+- property: action
+  ruby_type: Symbol
+  required: false
+  description_list:
+  - markdown: What action to execute. Defaults to :create if not specified.
```
Comment (IanMadd) [inline data/infra/resources/launchd.yaml]: Is this right?

File: data/infra/resources/batch.yaml (hunk 2 above)
Comment (johnmccrae) [reply data/infra/resources/batch.yaml]: corrected

File: data/infra/resources/batch.yaml (hunk 3 above)
Comment (johnmccrae) [reply data/infra/resources/batch.yaml]: corrected

File: data/infra/resources/bash.yaml (hunk 4 above)
Comment (johnmccrae) [reply data/infra/resources/bash.yaml:112]: These were removed as duplicates. They are also listed at the top of the document on lines 2 - 5

File: data/infra/resources/csh.yaml (hunk 6 above)
Comment (johnmccrae) [reply data/infra/resources/csh.yaml:64]: yes

File: data/infra/resources/cookbook_file.yaml (hunk 5 above)
Comment (johnmccrae) [reply data/infra/resources/cookbook_file.yaml]: corrected

File: data/infra/resources/csh.yaml (hunk 6 above)
Comment (johnmccrae) [reply data/infra/resources/csh.yaml]: updated

File: data/infra/resources/csh.yaml (hunk 7 above)
Comment (johnmccrae) [reply data/infra/resources/csh.yaml]: updated

File: data/infra/resources/csh.yaml (hunk 8 above)
Comment (johnmccrae) [reply data/infra/resources/csh.yaml]: So many mistakes in the docs... wow... updated

File: data/infra/resources/directory.yaml (hunk 9 above)
Comment (johnmccrae) [reply data/infra/resources/directory.yaml]: Corrected

File: data/infra/resources/dsc_resource.yaml (hunk 12 above)
Comment (johnmccrae) [reply data/infra/resources/dsc_resource.yaml]: updated

File: data/infra/resources/macosx_service.yaml [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+  :stop:
+    markdown: 
+  :restart:
+    markdown: 
+  :reload:
+    markdown: 
+  :mask:
+    markdown: 
+  :unmask:
+    markdown:
```
Comment (IanMadd) [inline data/infra/resources/macosx_service.yaml]: Is there text we can add to these?

File: data/infra/resources/macosx_service.yaml [hunk 21]
```diff
@@ -0,0 +1,174 @@
...
+  required: false
+  default_value: The resource block's name
+  description_list:
+  - markdown: An optional property to set the service name if it differs from the
+      resource block's name.
+- property: session_type
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown:
```
Comment (IanMadd) [inline data/infra/resources/macosx_service.yaml]: Is there text for this?

File: data/infra/resources/macports_package.yaml [hunk 22]
```diff
@@ -44,20 +44,18 @@ syntax_full_properties_list:
   available to this resource."
 actions_list:
   :install:
-    markdown: Default. Install a package. If a version is specified, install the specified
-      version of the package.
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
```
Comment (IanMadd) [inline data/infra/resources/macports_package.yaml:50]: Should this be deleted?

File: data/infra/resources/openbsd_package.yaml [hunk 23]
```diff
@@ -8,46 +8,33 @@ resource_description_list:
...
+- "`action` identifies which steps Chef Infra Client will take to bring the node into
+  the desired state."
+- "`options`, `package_name`, `source`, `timeout`, and `version` are the properties
+  available to this resource."
 actions_list:
   :install:
-    markdown: Default. Install a package. If a version is specified, install the specified
-      version of the package.
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
```
Comment (IanMadd) [inline data/infra/resources/openbsd_package.yaml:45]: Should this be removed?

File: data/infra/resources/pacman_package.yaml [hunk 24]
```diff
@@ -40,25 +28,23 @@ syntax_full_properties_list:
   available to this resource."
 actions_list:
   :install:
-    markdown: Default. Install a package. If a version is specified, install the specified
-      version of the package.
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
```
Comment (IanMadd) [inline data/infra/resources/pacman_package.yaml:46]: Should this be deleted?

File: data/infra/resources/perl.yaml [hunk 25]
```diff
@@ -8,100 +8,109 @@ resource_description_list:
     to the **execute** resource. Commands that are executed with this resource are
     (by their nature) not idempotent, as they are typically unique to the environment
     in which they are run. Use `not_if` and `only_if` to guard this resource for idempotence.
-syntax_description: "A **perl** resource block executes scripts Perl:\n\n```perl\n\
-  perl 'hello world' do\n  code <<-EOH\n    print \"Hello world! From Chef and Perl.\"\
-  ;\n  EOH\nend\n```"
+syntax_full_code_block: |-
+  perl 'name' do
+    code             String
```
Comment (IanMadd) [inline data/infra/resources/perl.yaml:26]: We've got a properties mismatch. These are the ones that I'm finding in one of these lists but not in the other:

command
default_env
domain
input
interpreter
elevated
ignore_failure
live_stream
login
password
notifies
subscribes
umask

File: data/infra/resources/portage_package.yaml [hunk 26]
```diff
@@ -27,20 +27,18 @@ syntax_full_properties_list:
   available to this resource."
 actions_list:
   :install:
-    markdown: Default. Install a package. If a version is specified, install the specified
+    markdown: (default) Install a package. If a version is specified, install the specified
       version of the package.
-  :nothing:
-    shortcode: resources_common_actions_nothing.md
```
Comment (IanMadd) [inline data/infra/resources/portage_package.yaml:33]: Should this be deleted?

File: data/infra/resources/powershell_package.yaml [hunk 27]
```diff
@@ -5,45 +5,48 @@ resources_common_notification: true
...
     markdown: Remove a package.
+  :purge:
+    markdown: Purge a package. This action typically removes the configuration files
+      as well as the package.
+  :reconfig:
+    markdown: Change the installed package
+  :lock:
+    markdown:
+  :unlock:
+    markdown:
```
Comment (IanMadd) [inline data/infra/resources/powershell_package.yaml]: Missing text.

File: data/infra/resources/powershell_script.yaml [hunk 28]
```diff
@@ -42,15 +56,15 @@ syntax_full_properties_list:
 - "`name` is the name given to the resource block."
 - "`action` identifies which steps Chef Infra Client will take to bring the node into
   the desired state."
-- "`code`, `command`, `convert_boolean_return`, `creates`, `cwd`, `domain`,
+- "`code`, `command`, `convert_boolean_return`, `creates`, `cwd`, `default_env`, `domain`,
   `elevated`, `environment`, `flags`, `group`, `input`, `interpreter`, `live_stream`,
-  `password`, `returns`, `timeout`, and `user` are the properties
-  available to this resource."
+  `login`, `password`, `returns`, `timeout`, and `user` are the properties available
+  to this resource."
```
Comment (IanMadd) [inline data/infra/resources/powershell_script.yaml]: We're missing default_env and login from the properites_list below

**Review by IanMadd — COMMENTED** [review]
More comments.

File: data/infra/resources/ruby.yaml [hunk 29]
```diff
@@ -17,36 +16,50 @@ syntax_description: |
     EOH
   end
   ```
+syntax_full_code_block: |-
+  ruby 'name' do
+    code             String
+    command          String, Array # default value: 'name' unless specified
```
Comment (IanMadd) [inline data/infra/resources/ruby.yaml]: In properties_list I'm not seeing: command, default_env, domain, elevated, input, interpreter, live_stream, login, password

These are in the properties_list but not the code block:
ignore_failure, notifies, umask

File: data/infra/resources/script.yaml [hunk 30]
```diff
@@ -65,169 +49,148 @@ syntax_properties_list:
...
-  \          String, Integer\n  interpreter                String\n  path        \
-  \               Array\n  returns                    Integer, Array\n  timeout  \
-  \                  Integer, Float\n  user                       String\n  password\
-  \                   String\n  domain                     String\n  umask       \
-  \               String, Integer\n  action                     Symbol # defaults\
-  \ to :run if not specified\nend"
+syntax_full_code_block: |-
+  script 'name' do
+    code             String
+    command          String, Array # default value: 'name' unless specified
```
Comment (IanMadd) [inline data/infra/resources/script.yaml]: In properties_list I'm not seeing: default_env, elevated, live_stream, login, password

These are in the properties_list but not the code block:
umask

File: data/infra/resources/script.yaml [hunk 31]
```diff
@@ -65,169 +49,148 @@ syntax_properties_list:
...
+  ruby_type: String, Array
   required: false
+  default_value: The resource block's name
   description_list:
-  - markdown: A quoted (" ") string of code to be executed.
+  - markdown: An optional property to set the command to be executed if it differs
+      from the resource block's name.
+  - note:
+      markdown: Use the **execute** resource to run a single command. Use multiple
+        **execute** resource blocks to run multiple commands.
```
Comment (IanMadd) [inline data/infra/resources/script.yaml]: Is this correct?

File: data/infra/resources/windows_service.yaml [hunk 32]
```diff
@@ -10,33 +10,50 @@ resource_description_list:
...
+syntax_full_code_block: |-
+  windows_service 'name' do
+    binary_path_name      String
+    delayed_start         true, false # default value: false
+    dependencies          String, Array
+    description           String
+    desired_access        Integer # default value: 983551
+    display_name          String
+    error_control         Integer # default value: 1
+    init_command          String
```
Comment (IanMadd) [inline data/infra/resources/windows_service.yaml]: These are listed here, but not included in properties_list:

init_command
options
parameters
priority
reload_command

**Review by IanMadd — COMMENTED** [review]
More questions for @johnmccrae 
This should be the last of it.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**